lib/Sema/CSApply.cpp

```diff
@@ -4465,6 +4465,43 @@ Expr *ExprRewriter::coerceExistential(Expr *expr, Type toType,
     return new (ctx) OpenExistentialExpr(expr, archetypeVal, result);
   }
 
+  // If the type we are trying to coerce is a tuple, let's look through
+  // its elements to see if there are any LValue types present, such requires
+  // load or address-of operation first before proceeding with erasure.
+  if (auto tupleType = fromType->getAs<TupleType>()) {
+    bool coerceToRValue = false;
+    for (auto element : tupleType->getElements()) {
+      if (element.getType()->is<LValueType>()) {
+        coerceToRValue = true;
+        break;
+      }
+    }
+
+    // Tuple has one or more LValue types associated with it,
```

Comment:
Actually, wait - it does look through the LValue but it won't modify AST to add implicit LoadExpr, so what we actually need here is coerceTupleToTuple which is going to process the expression and not only type.